### ISSUE TITLE
unable to use vue-loader 15.0.0 (Error: options/query provided without loader)

### ISSUE THREAD

**Issue by ilyaztsv** [storybooks/storybook#3492]
### Bug or support request summary

I've updated `"vue-loader": "^14.2.2"` to `"vue-loader": "^15.0.0"` in `examples/vue-kitchen-sink`. That update crashed the example for `4.0.0-alpha.3`.

### Steps to reproduce

- `cd examples/vue-kitchen-sink` (master branch)
- `npm i`
- `npm i --save-dev vue-template-compiler`
- `npm i --save-dev vue-loader@15.0.0`
- `npm run storybook`

You'll see errors like this:

```
ERROR in ./src/App.vue?vue&type=style&index=0&lang=css
Module parse failed: Unexpected character '#' (34:0)
You may need an appropriate loader to handle this file type.
|
|
| #app {
|   font-family: 'Avenir', Helvetica, Arial, sans-serif;
|   -webkit-font-smoothing: antialiased;
 @ ./src/App.vue 4:0-62
 @ ./src/stories/index.js
 @ ./.storybook/config.js
 @ multi ./node_modules/@storybook/vue/dist/server/config/polyfills.js ./node_modules/@storybook/vue/dist/server/config/globals.js (webpack)-hot-middleware/client.js?reload=true ./.storybook/config.js
```

If you add using of`VueLoaderPlugin` in `examples/vue-kitchen-sink/.storybook/webpack.config.js`

```
const path = require('path');
const { VueLoaderPlugin } = require('vue-loader');

module.exports = (storybookBaseConfig, configType, defaultConfig) => {
  defaultConfig.module.rules.push({
    test: [/\.stories\.js$/, /index\.js$/],
    loaders: [require.resolve('@storybook/addon-storysource/loader')],
    include: [path.resolve(__dirname, '../src')],
    enforce: 'pre',
  });

  defaultConfig.plugins.push(new VueLoaderPlugin());

  return defaultConfig;
};
```

and run `npm run storybook`, you'll see the following error:

```
/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/webpack/lib/RuleSet.js:262
			throw new Error(
			^

Error: options/query provided without loader (use loader + options) in {
  "query": {
    "cacheDirectory": "/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/.cache/react-storybook",
    "presets": [
      [
        "env",
        {
          "modules": false
        }
      ],
      "stage-0",
      "vue"
    ],
    "env": {
      "test": {
        "presets": [
          "env"
        ]
      }
    },
    "babelrc": false
  },
  "include": [
    "/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink"
  ],
  "exclude": [
    "/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules"
  ],
  "use": [
    {
      "options": {
        "cacheDirectory": "/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/.cache/react-storybook",
        "presets": [
          [
            "env",
            {
              "modules": false
            }
          ],
          "stage-0",
          "vue"
        ],
        "env": {
          "test": {
            "presets": [
              "env"
            ]
          }
        },
        "babelrc": false
      },
      "loader": "/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/babel-loader/lib/index.js"
    }
  ]
}
    at Function.normalizeRule (/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/webpack/lib/RuleSet.js:262:10)
    at rules.map (/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/webpack/lib/RuleSet.js:110:20)
    at Array.map (<anonymous>)
    at Function.normalizeRules (/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/webpack/lib/RuleSet.js:109:17)
    at new RuleSet (/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/webpack/lib/RuleSet.js:104:24)
    at new NormalModuleFactory (/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/webpack/lib/NormalModuleFactory.js:103:18)
    at Compiler.createNormalModuleFactory (/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/webpack/lib/Compiler.js:443:31)
    at Compiler.newCompilationParams (/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/webpack/lib/Compiler.js:460:30)
    at Compiler.compile (/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/webpack/lib/Compiler.js:468:23)
    at compiler.hooks.watchRun.callAsync.err (/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/webpack/lib/Watching.js:77:18)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:24:1)
    at AsyncSeriesHook.lazyCompileHook [as _callAsync] (/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/tapable/lib/Hook.js:35:21)
    at Watching._go (/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/webpack/lib/Watching.js:40:32)
    at Watching.compiler.readRecords.err (/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/webpack/lib/Watching.js:32:9)
    at Compiler.readRecords (/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/webpack/lib/Compiler.js:342:11)
    at new Watching (/Users/ilyaztsv/projects/my/storybook/examples/vue-kitchen-sink/node_modules/webpack/lib/Watching.js:29:17)
```

### Please specify which version of Storybook and optionally any affected addons that you're running

```
"dependencies": {
    "vue": "^2.5.16",
    "vuex": "^3.0.0"
  },
  "devDependencies": {
    "@storybook/addon-actions": "4.0.0-alpha.3",
    "@storybook/addon-centered": "4.0.0-alpha.3",
    "@storybook/addon-knobs": "4.0.0-alpha.3",
    "@storybook/addon-links": "4.0.0-alpha.3",
    "@storybook/addon-notes": "4.0.0-alpha.3",
    "@storybook/addon-options": "4.0.0-alpha.3",
    "@storybook/addon-storyshots": "4.0.0-alpha.3",
    "@storybook/addon-storysource": "4.0.0-alpha.3",
    "@storybook/addon-viewport": "4.0.0-alpha.3",
    "@storybook/addons": "4.0.0-alpha.3",
    "@storybook/vue": "4.0.0-alpha.3",
    "babel-core": "^6.26.0",
    "babel-loader": "^7.1.4",
    "babel-preset-env": "^1.6.0",
    "babel-preset-vue": "^2.0.2",
    "cross-env": "^5.1.4",
    "file-loader": "^1.1.11",
    "svg-url-loader": "^2.3.2",
    "vue-loader": "^15.0.0",
    "vue-template-compiler": "^2.5.16",
    "webpack": "^4.6.0",
    "webpack-dev-server": "^3.1.3"
  }
```

### Affected platforms 

- `"vue-loader": "^15.0.0"`

**Comment by ilyaztsv** [comment]
@Hypnosphi  do you have any news about fixing that issue?

**Comment by jaunkst** [comment]
+1

**Comment by Robula** [comment]
Same issue here, I'm extending webpack configuration:
```
const { VueLoaderPlugin } = require('vue-loader')

module.exports = {
  plugins: [new VueLoaderPlugin()]
}
```

**Comment by takaya1992** [comment]
FYI https://vue-loader.vuejs.org/migrating.html

**Comment by mpopv** [comment]
@Robula I believe in addition to adding VueLoaderPlugin you also need to rewrite the vue-loader options per the 15.0.0 migration docs, to something like
```js
      {
        test: /\.vue$/,
        loader: 'vue-loader',
        options: {
          loaders: {
            'scss': [
              'vue-style-loader',
              'css-loader',
              'sass-loader'
            ]
          }
        }
      },
```

**Comment by zhaozhe0831** [comment]
I had same problem. I spend half of the day on this devDependencies.
It based from @storybook/cli . The cli doesn't have the `vue-loader` of devDependencis.

**Comment by ilyaztsv** [comment]
@zhaozhe0831 could you show an example how to solve the issue?

**Comment by zhaozhe0831** [comment]
@ilyaztsv You can install below v15.0.0. 
like this `vue-loader`.
![image](https://user-images.githubusercontent.com/12945873/40338952-8cee1622-5daa-11e8-8314-6b6d8a148629.png)

**Comment by zhaozhe0831** [comment]
No, you need below v15.0.0.
You can use `npm list vue-loader` to check your `vue-loader` version.
You can install it with this code `npm install vue-loader@14.2.2`.
I think it’s useful. I’m not good at English. I’m not sure if not my text is correct .


原始邮件
发件人:Ilya Z.notifications@github.com
收件人:storybooks/storybookstorybook@noreply.github.com
抄送:阿炳zhaozhe0831@vip.qq.com; Mentionmention@noreply.github.com
发送时间:2018年5月22日(周二) 11:21
主题:Re: [storybooks/storybook] unable to use vue-loader 15.0.0 (Error:options/query provided without loader) (#3492)


@zhaozhe0831 thanks for the reply!
I wrote in the first issue message:
I've updated "vue-loader": "^14.2.2" to "vue-loader": "^15.0.0" in examples/vue-kitchen-sink.
but it did not solve the issue. Do you mean that I need update vue-loader in root folder of storybook (no only in examples/vue-kitchen-sink)?

**Comment by zhaozhe0831** [comment]
@ilyaztsv 
No, you need below v15.0.0.
You can use `npm list vue-loader` to check your `vue-loader` version.
You can install it with this code `npm install vue-loader@14.2.2`.
I think it’s useful. I’m not good at English. I’m not sure if not my text is correct .

**Comment by ilyaztsv** [comment]
@Hypnosphi are you planning to fix the issue or need PR?

**Comment by igor-dv** [comment]
@storybooks/team any reason why we can't update to v15?

**Comment by pksunkara** [comment]
Made #3911 to fix this.

**Comment by pksunkara** [comment]
@igor-dv Can we close this now?

**Comment by igor-dv** [comment]
Usually, we close issues after they were released. Until that, we mark them as `merged` =) It will be closed after the next alpha

**Comment by igor-dv** [comment]
I am sure as soon as @shilman will have a time =) 
Regarding the stable release - we have a few open PRs that we would like to integrate as v4. 

@ndelangen, any insights?

**Comment by Hypnosphi** [comment]
Released as (`4.0.0-alpha.15`)[https://github.com/storybooks/storybook/releases/tag/v4.0.0-alpha.15]
